libavformat/movenc.c

```diff
@@ -6208,23 +6213,31 @@ static int mov_init(AVFormatContext *s)
         return AVERROR(ENOMEM);
 
     if (mov->encryption_scheme_str != NULL && strcmp(mov->encryption_scheme_str, "none") != 0) {
-        if (strcmp(mov->encryption_scheme_str, "cenc") == 0) {
-            mov->encryption_scheme = MOV_ENC_CENC;
+        if (strcmp(mov->encryption_scheme_str, "cenc-aes-ctr") == 0) {
+            mov->encryption_scheme = MOV_ENC_CENC_AES_CTR;
             if (mov->encryption_key_len != AES_CTR_KEY_SIZE) {
                 av_log(s, AV_LOG_ERROR, "Invalid encryption key len %d expected %d\n",
                     mov->encryption_key_len, AES_CTR_KEY_SIZE);
                 return AVERROR(EINVAL);
             }
-            if (mov->encryption_kid_len != CENC_KID_SIZE) {
-                av_log(s, AV_LOG_ERROR, "Invalid encryption kid len %d expected %d\n",
-                    mov->encryption_kid_len, CENC_KID_SIZE);
+        } else if (strcmp(mov->encryption_scheme_str, "cenc-aes-cbc-pattern") == 0) {
+            mov->encryption_scheme = MOV_ENC_CENC_AES_CBC_PATTERN;
+            if (mov->encryption_key_len != 16 && mov->encryption_key_len != 24 && mov->encryption_key_len != 32) {
```

Comment:
Where is encryption_key_len set? When encryption_key_len can be 24 or 32? The encryption scheme is the same but key length is different, how?

Reply:
I think I got it from the IETF CBC spec: https://tools.ietf.org/html/rfc3602#section-2.2

The CENC spec says 128 bit for both CTR and CBC so I updated